From 363730f5ab307096cbb3873af0f576be0406cf46 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 242/296] Fix for wallpaper flicker on bootup on some targets

We see wallpaper flicker in Android N on some devices. Issue
happens due to wallpaper target keyguard Scrim being destroyed
and new wallpaper target statusbar coming up. This creates a
surface with width & height larger than the device supported
width and height. This causes a momentary jank in UI during
bootup.
---
 packages/SystemUI/src/com/android/systemui/ImageWallpaper.java | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java b/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java
index 8f3f6bf4b5c..bebe81cccd4 100644
--- a/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java
+++ b/packages/SystemUI/src/com/android/systemui/ImageWallpaper.java
@@ -500,9 +500,6 @@ public class ImageWallpaper extends WallpaperService {
                     }
                     updateSurfaceSize(getSurfaceHolder(), getDefaultDisplayInfo(),
                             false /* forDraw */);
-                    if (mNeedsDrawAfterLoadingWallpaper) {
-                        drawFrame();
-                    }
 
                     mLoader = null;
                     mNeedsDrawAfterLoadingWallpaper = false;
-- 
2.11.0

